samples/06.recomposing-ui/a.minimizable-web-chat/README.md

```diff
@@ -72,181 +76,155 @@ Open the project in your preferred IDE.
+In the `WebChat.js` file, import `React`, `{ useEffect, useMemo }` `ReactWebChat`, `createDirectLine`, and `createStyleSet` from our packages.
 
 <!-- prettier-ignore-start -->
 ```js
-import memoize from 'memoize-one';
 import React from 'react';
 import ReactWebChat, { createDirectLine, createStyleSet } from 'botframework-webchat';
 ```
 <!-- prettier-ignore-end -->
 
-Set up the component the same way you would set up a regular `React.Component`
+Set up the function component the same way you would set up a regular `React.Component`
 
-1. Create the `constructor()` and `render()` methods.
-1. In the constructor you can set state and `createDirectLine` with your token. It is **never recommended** to put the Direct Line secret in the browser or client app. To learn more about secrets and tokens for Direct Line, visit this [tutorial on authentication](https://docs.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-direct-line-3-0-authentication).
+1. Set state with `useMemo` and `createDirectLine` with your token. It is **never recommended** to put the Direct Line secret in the browser or client app. To learn more about secrets and tokens for Direct Line, visit this [tutorial on authentication](https://docs.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-direct-line-3-0-authentication).
 1. Inside render, be sure to return the `<ReactWebChat>` component.
 
 <!-- prettier-ignore-start -->
 ```js
-constructor(props) {
-  super(props);
-
-  this.createDirectLine = memoize(token => createDirectLine({ token }));
-
-  this.state = {
-    styleSet: createStyleSet({
-      backgroundColor: 'Transparent'
-    })
-  };
-}
+const WebChat = ({ className, onFetchToken, store, token }) => {
+  const directLine = useMemo(() => createDirectLine({ token }), [token]);
 
-render() {
-  const {
-    props: { className, store, token },
-    state: { styleSet }
-  } = this;
+  useEffect(() => {
+    onFetchToken();
+  }, [onFetchToken]);
 
-  return (
-    <ReactWebChat
-      className={ `${ className || '' } web-chat` }
-      directLine={ this.createDirectLine(token) }
-      store={ store }
-      styleSet={ styleSet }
-    />
+  return token ? (
+    <ReactWebChat className={`${className || ''} web-chat`} directLine={directLine} store={store} styleSet={styleSet} />
+  ) : (
+    <div className={`${className || ''} connect-spinner`}>
+      <div className="content">
+        <div className="icon">
+          <span className="ms-Icon ms-Icon--Robot" />
+        </div>
+        <p>Please wait while we are connecting.</p>
+      </div>
+    </div>
   );
-}
+};
 ```
 <!-- prettier-ignore-end -->
 
-Create the React method `componentDidMount` and invoke `onFetchToken` from props if the `token` has not been passed from the parent component.
+With `useEffect`, invoke `onFetchToken` from props if the `token` has not been passed from the parent component.
```

Comment:
```suggestion
In the `useEffect` hook, invoke `onFetchToken` from props if the `token` has not been passed from the parent component.
```

Reply:
This isn't really clear because the logic to check if the token has been fetched is in the `onFetchToken` - not the in the `useEffect` hook.